mdn/kuma
# Wire up SubscriptionForm

Fixes #6728

## What it does
- adds an API endpoint for creating subscriptions
  - basically the same as `/profile/stripe_subscription` minus the redirecting
  - uses the same `create_stripe_customer_and_subscription_for_user` function
    - @peterbe I moved your `UserSubscription` call into that function, so that I don't have to repeat it
- uses that new endpoint in SubscriptionForm
- prefills stripe modal with user's email
- adds an effect hook function that conditonally loads a script, so that the stripe modal script is only conditionally loaded once a user checks the box

## What it does not do
This does not bring the view into a state ready for users as it redirects to a not (yet) existing thank-you page (#6717). It also still shows the subscription form even if the user has already subscribed. The new SubHeader data also depends on #6717 landing.

## How to test it
Delete your existing subscription
http://localhost.org:8000/en-US/payments/recurring/management

Set up a new one
http://localhost.org:8000/en-US/payments/

Validate that it navigates to `thank-you` (as in the URL changes, no visual changes expected) and that your new subscription shows up in the management page.
